src/core/irc_client.ml

```diff
@@ -1,5 +1,7 @@
 module Log = Irc_helpers.Log
 
+type sasl = [`None | `Plain | `External]
```

Comment:
I would prefer ```type sasl = [`Plain | `External]``` here and then `?sasl:sasl option` in the `connect`/`connect_by_name` functions. Otherwise this type is kinda reinventing `option` :)

Reply:
Good point, thanks.